components/form/form.js

```diff
@@ -0,0 +1,197 @@
+import '../tooltip/tooltip.js';
+import { findFormElements, getFormElementData, isCustomFormElement, isNativeFormElement } from './form-helper.js';
+import { html, LitElement } from 'lit-element/lit-element.js';
+import { getComposedActiveElement } from '../../helpers/focus.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { LocalizeCoreElement } from '../../lang/localize-core-element.js';
+import { localizeFormElement } from './form-element-localize-helper.js';
+
+class Form extends LocalizeCoreElement(LitElement) {
+
+	static get properties() {
+		return {
+			action: { type: String },
+			method: { type: String },
+			trackChanges: { type: Boolean, attribute: 'track-changes', reflect: true }
+		};
+	}
+
+	constructor() {
+		super();
+		this._onUnload = this._onUnload.bind(this);
+		this._onNativeSubmit = this._onNativeSubmit.bind(this);
+
+		this.trackChanges = false;
+		this._tooltips = new Map();
+		this._validationCustoms = new Set();
+
+		this.addEventListener('d2l-validation-custom-connected', this._validationCustomConnected);
+		this.addEventListener('d2l-validation-custom-disconnected', this._validationCustomDisconnected);
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		window.addEventListener('beforeunload', this._onUnload);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.removeEventListener('beforeunload', this._onUnload);
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		this.addEventListener('change', this._onFormElementChange);
+		this.addEventListener('input', this._onFormElementChange);
+		this.addEventListener('focusout', this._onFormElementChange);
+
+		this._form = document.createElement('form');
+		this._form.addEventListener('submit', this._onNativeSubmit);
+		this._form.id = getUniqueId();
+		this._form.noValidate = true;
+		this.appendChild(this._form);
+
+		const formElements = findFormElements(this);
```

Comment:
What happens if custom/native form elements change after this runs... like if they're added/removed from the DOM.

Reply:
Unfortunately currently won't be detected. If we want to support that behavior then we'll need to use a mutation observer. I suspect we'll have to go that direction due to conditional rendering but wanted to keep it simple in this PR.

Reply:
Cool. Yeah I imagine this will definitely come up in more complex apps where they want a top-level form but then conditionally render different things in sub-views. We can figure that out later.